testsuite/tests/parsetree/source.ml

```diff
@@ -171,11 +171,13 @@ and[@foo] y = x
 type%foo[@foo] t = int
 and[@foo] t = int
 type%foo[@foo] t += T
+type t += A = M.A[@a]
```

Comment:
I might be nice to add the more convoluted example:
``ocaml
type t += B = M.A[@b] | C = M.A[@c][@@t]
```